drivers/sensor/sensirion/scd4x/scd4x.c

```diff
@@ -40,18 +40,81 @@ struct scd4x_data {
     int32_t humidity;
 };
 
+/* scd4x_read */
+static int scd4x_read(const struct device *dev, uint16_t cmd, size_t len, uint16_t *val)
+{
+    int ret;
+    const struct scd4x_config *config = dev->config;
+    static char read_buf[3 * MAX_READ_SIZE];
+
+    U16_TO_BYTE_ARRAY(cmd, read_buf);
+    ret = i2c_write_dt(&config->i2c, (const uint8_t *)&read_buf, sizeof(cmd));
+    if (ret < 0) {
+        LOG_ERR("Failed to write command 0x%04x, error %d", cmd, ret);
+        return ret;
+    }
+
+    k_sleep(K_MSEC(1));
+
+    ret = i2c_read_dt(&config->i2c, read_buf, len * 3);
+    if (ret < 0) {
+        LOG_ERR("Failed to read response");
+        return ret;
+    }
+
+    uint8_t crc;
+    for (int i = 0; i < len; i++) {
+        crc = crc8(read_buf + (3 * i), 2, CRC8_POLYNOMIAL, CRC8_INIT, false);
+        if (crc != read_buf[3 * i + 2]) {
+            LOG_ERR("CRC error");
+            return -EIO;
+        }
+        BYTE_ARRAY_TO_U16((read_buf + (3 * i)), val[i]);
+    }
+
+    return 0;
+}
+
+/* scd4x_write */
+static int scd4x_write(const struct device *dev, uint16_t cmd, size_t len, uint16_t *val)
+{
+    int ret;
+    const struct scd4x_config *config = dev->config;
+    static char write_buf[sizeof(cmd) + MAX_READ_SIZE * 3];
+
+    U16_TO_BYTE_ARRAY(cmd, write_buf);
+    for (int i = 0; i < len; i++) {
+        U16_TO_BYTE_ARRAY(val[i], (write_buf + (3 * i) + sizeof(cmd)));
+        write_buf[3 * i + 2 + sizeof(cmd)]
+                = crc8(write_buf + (3 * i) + sizeof(cmd), 2, CRC8_POLYNOMIAL, CRC8_INIT, false);
+    }
+
+    ret = i2c_write_dt(&config->i2c, (char *)&write_buf, sizeof(cmd) + len * 3);
+    if (ret < 0) {
+        LOG_ERR("Failed to write command 0x%04x, error %d", cmd, ret);
+        return ret;
+    }
+
+    return 0;
+}
+
+/* scd4x_send_command */
+static int scd4x_send_command(const struct device *dev, uint16_t cmd)
+{
+    return scd4x_write(dev, cmd, 0, NULL);
+}
+
+/* scd4x_sample_fetch */
 static int scd4x_sample_fetch(const struct device *dev, enum sensor_channel chan)
 {
     struct scd4x_data *data = dev->data;
 
-    // Check if sensor is ready
     uint16_t ready;
     scd4x_read(dev, SCD4X_CMD_GET_DATA_READY_STATUS, 1, &ready);
     if ((ready & 0x03FF) == 0) {
         return -EBUSY;
     }
 
```

Comment:
Comment format will be addressed in an other issue.

```suggestion

    // Read measurements
```